### PR TITLE
Use word index for determining whether a token is a subword when addi…

### LINKED ISSUE #17220: LayoutLMv2 Fast Tokenizer improperly aligns labels for non-first subwords with 0 offsets
### System Info

```shell
Copy-and-paste the text below in your GitHub issue and FILL OUT the two last points.

- `transformers` version: 4.16.0
- Platform: macOS-12.3.1-x86_64-i386-64bit
- Python version: 3.8.9
- PyTorch version (GPU?): 1.8.1 (False)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>
```

### Who can help?

 @NielsRogge

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

```
In [1]: from transformers import LayoutLMv2TokenizerFast
None of PyTorch, TensorFlow >= 2.0, or Flax have been found. Models won't be available and only tokenizers, configuration and file/data utilities can be used.

In [2]: tokenizer = LayoutLMv2TokenizerFast.from_pretrained("microsoft/layoutlmv2-base-uncased")

In [4]: toks = tokenizer(['그'], boxes=[[1,2,3,4]], word_labels=[2])

In [5]: toks
Out[5]: {'input_ids': [101, 1455, 30017, 102], 'token_type_ids': [0, 0, 0, 0], 'attention_mask': [1, 1, 1, 1], 'bbox': [[0, 0, 0, 0], [1, 2, 3, 4], [1, 2, 3, 4], [1000, 1000, 1000, 1000]], 'labels': [-100, 2, 2, -100]}

In [9]: toks.labels
Out[9]: [-100, 2, 2, -100]

```

### Expected behavior

```shell
Since the single word '그' was split into 2 tokens, the second subword should be assigned a -100 value.

This is happening because of logic here:

https://github.com/huggingface/transformers/blob/3f936df66287f557c6528912a9a68d7850913b9b/src/transformers/models/layoutlmv2/tokenization_layoutlmv2_fast.py#L599


where a token's offset is used to determine whether it is the first subword or not.

The problem with '그' is that it is split into 2 subword: 'ᄀ', '##ᅳ' that share an offset of (0,1) so both given the label of the word.

By using solely the word_index to decide which is a subword this problem can be avoided.
```

### PR DESCRIPTION
**Pull request by timothyjlaurent**
…ng word labels.

# What does this PR do?
Use repeated word index instead of offset[0] == 0 for assigning pad labels to non-first subwords.

As described in the #17220, words like 그 will be split into 2 subwords both with 0,1 offsets but with the same word index. This PR ensures that this second subword receives the correct -100 label

Fixes #17220

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?
I tried to write a test but the test tokenizer doesn't split my test word. Open to suggestions or maybe it can be a slow test with the pretrained tokenizer.

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@NielsRogge

### CODE REVIEW

**Comment by timothyjlaurent** [comment]
@NielsRogge  do you have suggestions for making a test case for the fast tokenizer that will split the "그" token?

Could I use the pretrained Layoutlmv2 tokenizer and mark the test as slow, or do you have suggestions for how to properly configure a test tokenizer that will split that token?

File: src/transformers/models/layoutlmv2/tokenization_layoutlmv2_fast.py [hunk 1]
```diff
@@ -581,16 +581,17 @@ def _batch_encode_plus(
                 else:
                     original_index = batch_index
                 labels_example = []
-                for id, offset, word_id in zip(
+                cur_word_id = -1
```
Comment (NielsRogge) [inline src/transformers/models/layoutlmv2/tokenization_layoutlmv2_fast.py:584]: Hi, what does `cur_word_id` mean? We'd like to use variable names that are easy to understand.

Comment (timothyjlaurent) [reply src/transformers/models/layoutlmv2/tokenization_layoutlmv2_fast.py:584]: It mean "current_word_id", @NielsRogge, I can update the name.
